apis/v1beta1/validation/gateway_test.go

```diff
@@ -66,6 +66,18 @@ func TestValidateGateway(t *testing.T) {
 			},
 			expectErrsOnFields: []string{"spec.listeners[0].tls"},
 		},
+		"tls config not set with https protocol": {
+			mutate: func(gw *gatewayv1b1.Gateway) {
+				gw.Spec.Listeners[0].Protocol = gatewayv1b1.HTTPSProtocolType
+			},
+			expectErrsOnFields: []string{"spec.listeners[0].tls"},
+		},
+		"tls config not set with tls protocol": {
+			mutate: func(gw *gatewayv1b1.Gateway) {
+				gw.Spec.Listeners[0].Protocol = gatewayv1b1.TLSProtocolType
+			},
+			expectErrsOnFields: []string{"spec.listeners[0].tls"},
+		},
```

Comment:
For completeness, please ad a test that validates all other protocols _do not_ get an error for this, then LGTM.

Reply:
Updated, thanks.